src/ManageCourses.ApiClient/HttpClientWrapper.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace GovUk.Education.ManageCourses.ApiClient
+{
+    public class HttpClientWrapper : IHttpClient
+    {
+        private readonly HttpClient wrapped;
+
+        public HttpClientWrapper(HttpClient wrapped)
+        {
+            if(wrapped == null)
+            {
+                throw new ManageCoursesApiException($"Failed to instantiate due to HttpClient = null");
+            }
+
+            this.wrapped = wrapped;
+        }
+
+        public async Task<HttpResponseMessage> GetAsync(Uri queryUri)
+        {
+            var msg = $"API GET Failed uri {queryUri}";
+            try
+            {
+                var result = await wrapped.GetAsync(queryUri);
+
```

Comment:
Can you not just use result.EnsureSuccessStatusCode() to throw an exception for you?

Reply:
Nope, we need to fish out the status code to throw the exception